views.py

```diff
@@ -64,34 +66,37 @@ def download(request, path, *args, **kwargs):
         response.content = "<h1>You do not have permission to download this resource!</h1>"
         return response
 
+    if is_bag_download:
+        # do on-demand bag creation
+        bag_modified = "false"
+        # needs to check whether res_id collection exists before getting/setting AVU on it to accommodate the case
+        # where the very same resource gets deleted by another request when it is getting downloaded
+        if federated_path:
+            if federated_path.endswith('/'):
+                res_root = '{}{}'.format(federated_path, res_id)
+            else:
+                res_root = '{}/{}'.format(federated_path, res_id)
+        else:
+            res_root = res_id
+        if istorage.exists(res_root):
+            bag_modified = istorage.getAVU(res_root, 'bag_modified')
+        if bag_modified == "true":
+            task = create_bag_by_irods.apply_async((res_id, istorage),
+                                                  countdown=3)
+            if rest_call:
+                return HttpResponse({'bag_status': 'Not ready',
```

Comment:
@hyi Test is failing due to this response. Change it to:

```JsonResponse()``` or use json.dumps() to pass the dict.